src/poly/commitment.rs

```diff
@@ -273,8 +433,40 @@ fn test_opening_proof() {
             transcript.absorb(Field::one());
         } else {
             let opening_proof = opening_proof.unwrap();
-            assert!(opening_proof.verify(&params, &mut transcript_dup, x, &p, v));
+            // Verify the opening proof
+            let msm = params.msm();
+            let guard = opening_proof
```

Comment:
One way to improve this test so it can catch the bug currently in `MSM::compute_g` is to call `use_challenges` on this guard to get a new msm, and use that msm to verify the same `opening_proof` again and get a new guard. That guard can then proceed with the rest of the test as it's currently written. This way the `g_scalars` field of the `MSM` is populated already and you can see that your changes to `MSM::compute_g` fixed the issue.